lib/heimdall_tools/aws_config_mapper.rb

```diff
@@ -0,0 +1,317 @@
+                              'code_desc': NOT_APPLICABLE_MSG,
+                              'skip_message': NOT_APPLICABLE_MSG,
+                              'start_time': DateTime.now.strftime('%Y-%m-%dT%H:%M:%S%:z'),
+                              'status': 'skipped'
+                            }
+                          when 'INSUFFICIENT_DATA'
+                            {
+                              'run_time': 0,
+                              'code_desc': INSUFFICIENT_DATA_MSG,
+                              'skip_message': INSUFFICIENT_DATA_MSG,
+                              'start_time': DateTime.now.strftime('%Y-%m-%dT%H:%M:%S%:z'),
+                              'status': 'skipped'
+                            }
+                          end
+      end
+
+      config_rules
+    end
+
+    ##
+    # Takes in the tags for a rule on AWS and pulls out tags that are useful for HDF.
+    #
+    # `csc_control` is currently unsupported.
+    #
+    # AWS Tag Rules: https://docs.aws.amazon.com/config/latest/developerguide/tagging.html
+    # - max 50 tags per resource
+    # - max 128 chars for keys
+    # - max 256 chars for values
+    # - chars limited to: Unicode letters, digits, whitespace, or _ . : / = + - @
+    #
+    # Tags where (key: example AWS value):
+    # {
+    #   'cce_id': 'CCE-78904-0'
+    #   'check': 'Perform the following...'
+    #   'cis_impact': 'This feature can be used...'
+    #   'cis_level': '1'
+    #   'cis_rid': '1.6'
+    #   'fix': 'Perform the following...'
+    #   'nist': 'IA-05-01:Rev_4:AU-02'
+    #   'rationale': 'Setting a...',
+    #   'cweid': ''
+    #   'confidence': ''
+    # }
+    #
+    # Would be returned as:
+    # {
+    #   'cce_id': 'CCE-78904-0',
+    #   'check': 'Perform the following...',
+    #   'cis_impact': 'This feature can be used...',
+    #   'cis_level': 1,
+    #   'cis_rid': '1.6',
+    #   'fix': 'Perform the following...',
+    #   'nist': [
+    #       'IA-5(1)',
+    #       'Rev_4',
+    #       'AU-2'
+    #   ],
+    #   'rationale': 'Setting a...',
+    #   'cweid': ''
+    #   'confidence': ''
+    # }
+    #
+    # Params:
+    # - aws_tags: The tags hash produces for a rule in AwsConfigMapper::add_tags_to_config_rules
+    #
+    # Returns: Hash containing all relevant HDF tags
+    def hdf_tags(aws_tags)
+      result = {}
+      # Passthrough tags
+      result['cce_id'] = aws_tags['cce_id'] if aws_tags.key? 'cce_id'
+      result['check'] = aws_tags['check'] if aws_tags.key? 'check'
+      result['cis_impact'] = aws_tags['cis_impact'] if aws_tags.key? 'cis_impact'
+      result['cis_rid'] = aws_tags['cis_rid'] if aws_tags.key? 'cis_rid'
+      result['fix'] = aws_tags['fix'] if aws_tags.key? 'fix'
+      result['rationale'] = aws_tags['rationale'] if aws_tags.key? 'rationale'
+      result['cweid'] = aws_tags['cweid'] if aws_tags.key? 'cweid'
+      result['confidence'] = aws_tags['confidence'] if aws_tags.key? 'confidence'
```

Comment:
Could you make passthrough_tags an array here and just do 
```
passthrough_tags.each do |tag|
    result[tag] = aws_tags[tag] if aws_tags.key? tag
end
```

Reply:
good idea

Reply:
Please see update to this section of code